File: Codes/C++/1. single-responsibility-principle.cpp
#include <iostream>
#include <fstream>
#include <string>

class Report {
public:
    std::string content;

    Report(const std::string& text) : content(text) {}

    std::string getContent() const {
        return content;
    }
};

class ReportPrinter {
public:
    void print(const Report& report) {
        std::cout << report.getContent() << std::endl;
    }
};

class ReportSaver {
public:
    void saveToFile(const Report& report, const std::string& filename) {
        std::ofstream file(filename);
        file << report.getContent();
        file.close();
    }
};

File: Codes/Dart/1. single-responsibility-principle.dart
import 'dart:io';

class Report {
  final String content;

  Report(this.content);

  String getContent() {
    return content;
  }
}

class ReportPrinter {
  void printReport(Report report) {
    print(report.getContent());
  }
}

class ReportSaver {
  void saveToFile(Report report, String filename) {
    final file = File(filename);
    file.writeAsStringSync(report.getContent());
  }
}

void main() {
  // Example usage:
  Report report = Report("This is a sample report.");
  ReportPrinter printer = ReportPrinter();
  ReportSaver saver = ReportSaver();

  printer.printReport(report);
  saver.saveToFile(report, 'report.txt');
}
